util/slicesx/slicesx.go

```diff
@@ -42,3 +42,16 @@ func Shuffle[S ~[]T, T any](s S) {
 		s[i], s[j] = s[j], s[i]
 	}
 }
+
+// Partition returns two slices, the first containing the elements of the input
+// slice for which the callback evaluates to true, the second containing the rest.
+func Partition[S ~[]T, T any](s S, cb func(T) bool) (a, b S) {
```

Comment:
maybe call the results `trues, falses S` ?

and document that it doesn't mutate s?